gcovr/__main__.py

```diff
@@ -152,6 +152,20 @@ def main(args=None):
 
     logger = Logger(options.verbose)
 
+    if sys.version_info < (3, 8) and options.use_canonical_paths:
+        logger.error("--use_canonical_paths can't be set due to incompatible Python version.")
+        sys.exit(1)
+
+    if options.use_canonical_paths:
+        def canonical_path(path: str, option: str) -> str:
+            canonical = os.path.realpath(path)
+            if canonical != path:
+                logger.msg(f"{option} has been normalized to {canonical}")
```

Comment:
Tests seem to be failing because of this line.

```suggestion
                logger.verbose_msg(f"{option} has been normalized to '{canonical}'")
```